src/promptflow-azure/promptflow/azure/_storage/cosmosdb/summary.py

```diff
@@ -127,6 +136,56 @@ def _persist_running_item(self, client: ContainerProxy):
             item.line_number = attributes[SpanAttributeFieldName.LINE_NUMBER]
         safe_create_cosmosdb_item(client, item)
 
+    # Parse inputs and outputs from span events and truncate the content for cosmosdb 2MB limit.
+    # After truncation, the query ability for inputs and outputs will be limited.
+    def _parse_inputs_outputs_from_events(self):
+        events = self.span.events
+        # Only use the first payload for input and output in case there are multiple valid events.
+        for event in events:
+            value_for_input = self._get_pf_defined_payload(event, SPAN_EVENTS_NAME_PF_INPUTS)
+            if value_for_input is not None:
+                self.inputs = self._truncate_inputs_outputs_content(value_for_input)
+                break
+        for event in events:
+            value_for_output = self._get_pf_defined_payload(event, SPAN_EVENTS_NAME_PF_OUTPUT)
+            if value_for_output is not None:
+                self.outputs = self._truncate_inputs_outputs_content(value_for_output)
+                break
+
+    def _get_pf_defined_payload(self, event, key):
+        if event.get(SpanEventFieldName.NAME, "") != key:
+            return None
+        attributes = event.get(SpanEventFieldName.ATTRIBUTES, {})
+        if SPAN_EVENTS_ATTRIBUTE_PAYLOAD not in attributes:
+            return None
+        #  Do not constraint the payload to be dict in case of trace from customer script.
+        return json_loads_parse_const_as_str(attributes[SPAN_EVENTS_ATTRIBUTE_PAYLOAD])
+
+    def _truncate_inputs_outputs_content(self, content):
+        def _process_value(value):
+            # For python, bool is subclass of int, so we don't need to check bool again.
+            if value is None or isinstance(value, (int, float)):
+                return value
+            elif isinstance(value, str):
+                # Truncate string values, use large enough limit for UX display.
+                return value[:500]
```

Comment:
nit: make this 500 a constant, in case we might change this in the future (you know this might happen...)

Reply:
also, for below hard code lines, might be better

Reply:
Fixed~